### PR TITLE
add pretty printing and error handling to ip_getter

### LINKED ISSUE #195: The error message when IP_GETTER fails is cryptic
Lets add error handling instead :)

Fallback handling would be nice too.

### PR DESCRIPTION
**Pull request by swalladge**
fixes #195

### CODE REVIEW

File: piqueserver/server.py [hunk 1]
```diff
@@ -347,15 +348,23 @@ def __init__(self, interface, config):
...
         try:
             ip = yield self.getPage(ip_getter)
-        except OSError as e:
+        except Exception as e:
             print("Getting external IP failed:", e)
             return
 
-        self.ip = ip.strip()
+        ip = ip.strip()
+        if not re.match(r'^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$', ip):
```
Comment (NotAFile) [inline piqueserver/server.py]: I'd prefer just putting this into ip_address to outsource the validation.

Might also be worth making `protocol.ip` an IPV4Address instance... but that might cause backwards-incompatibility. Perhaps something along the lines of renaming it to `ip_address` or something and then making `ip` a property that returns `str(self.ip_address)`? The places where self.ip are used will be very limited though, so I don't think it's at all worth it.

Comment (swalladge) [reply piqueserver/server.py]: :+1:

I don't think protocol.ip is actually used anywhere though.

File: piqueserver/server.py [hunk 2]
```diff
@@ -351,19 +350,19 @@ def get_external_ip(self, ip_getter):
...
                   'Please check the "ip_getter" setting in your config.')
             return
+        except Exception as e:
+            print("Getting external IP failed:", e)
+            return
 
-        self.ip = ip
-        self.identifier = make_server_identifier(ip, self.port)
-        print('Server public ip address: {}:{}'.format(ip, self.port))
+        self.ip = ip.exploded
```
Comment (NotAFile) [inline piqueserver/server.py]: The thing here is that ip and ip_address might go out of sync...

Comment (NotAFile) [reply piqueserver/server.py]: Also, I just looked it up and no code ever accesses `*.ip*`

**Pull request merged**